tests/tests.rs

```diff
@@ -5,42 +5,132 @@ extern crate shlex;
 
 use bindgen::{Builder, builder, clang_version};
 use std::fs;
-use std::io::{BufRead, BufReader, Error, ErrorKind, Read, Write};
+use std::io::{self, BufRead, BufReader, Error, ErrorKind, Read, Write};
 use std::path::PathBuf;
+use std::process;
+use std::sync::{Once, ONCE_INIT};
 
 #[path = "../src/options.rs"]
 mod options;
 use options::builder_from_flags;
 
+// Run `rustfmt` on the given source string and return a tuple of the formatted
+// bindings, and rustfmt's stderr.
+fn rustfmt(source: String) -> (String, String) {
+    static INSTALL_RUSTFMT: Once = ONCE_INIT;
+
+    INSTALL_RUSTFMT.call_once(|| {
+        let have_working_rustfmt = process::Command::new("rustup")
+            .args(&["run", "nightly", "rustfmt", "--version"])
+            .stdout(process::Stdio::null())
+            .stderr(process::Stdio::null())
+            .status()
+            .expect("should run `rustup run nightly rustfmt --version` OK")
+            .success();
+
+        if have_working_rustfmt {
+            return;
+        }
+
+        // Because `rustfmt` needs to match its exact nightly version, we update
+        // both at the same time.
+
+        let status = process::Command::new("rustup")
```

Comment:
It definitely feels wrong to update the nightly in behalf of the user.

Reply:
Unfortunately, this is the only way for the latest `rustfmt` to work :-/